File: src/main/java/com/example/authapp/dto/response/UserResponse.java
package com.example.authapp.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {
    private Long id;
    private String name;
    private String email;
    private String provider;
    private String avatarUrl;
    private Set<String> roles;
    private String dateOfBirth;
    private Boolean salesNotification;
    private Boolean newArrivalsNotification;
    private Boolean deliveryStatusNotification;
}

File: frontend/lib/features/auth/models/user_response.dart
class UserResponse {
  final int id;
  final String name;
  final String email;
  final String provider;
  final String? avatarUrl;
  final List<String> roles;
  final String? dateOfBirth;
  final bool salesNotification;
  final bool newArrivalsNotification;
  final bool deliveryStatusNotification;

  UserResponse({
    required this.id,
    required this.name,
    required this.email,
    required this.provider,
    this.avatarUrl,
    required this.roles,
    this.dateOfBirth,
    this.salesNotification = true,
    this.newArrivalsNotification = true,
    this.deliveryStatusNotification = true,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      provider: json['provider'] ?? 'local',
      avatarUrl: json['avatarUrl'],
      roles: json['roles'] != null ? List<String>.from(json['roles']) : [],
      dateOfBirth: json['dateOfBirth'],
      salesNotification: json['salesNotification'] ?? true,
      newArrivalsNotification: json['newArrivalsNotification'] ?? true,
      deliveryStatusNotification: json['deliveryStatusNotification'] ?? true,
    );
  }
}
